src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/TrivialTemporaryStorageService.cs

```diff
@@ -94,10 +93,8 @@ public Task<SourceText> ReadTextAsync(CancellationToken cancellationToken = defa
 
             public void WriteText(SourceText text, CancellationToken cancellationToken = default(CancellationToken))
             {
-                // Decompose the SourceText into it's underlying parts, since we use it as a key
-                // into many other caches that don't expect it to be held
-                _text = text.ToString();
-                _encoding = text.Encoding;
+                // This is a trivial implementation, indeed.
+                _sourceText = text;
```

Comment:
Actually, scratch that. MemberwiseClone is protected, so it can't be called from the outside. We'd need to introduce a virtual Clone method on SourceText (a bit late for that now).

Another thought I had was to compress the text using System.IO.Compression.DeflateStream.